### PR TITLE
Fixes #21789: enable select all on errata page.

### PR DESCRIPTION
**Pull request by waldenraines**
http://projects.theforeman.org/issues/21789

### CODE REVIEW

**Comment by jlsherrill** [comment]
This isn't working quite as expected.  The search string sent to the server is "" which match all hosts, not just the hosts that are in the list.  The list should be using a search string such as "( applicable_errata = "RHBA-2017:2924" )"  and this should send that too.

**Comment by waldenraines** [comment]
> This isn't working quite as expected. The search string sent to the server is "" which match all hosts, not just the hosts that are in the list. The list should be using a search string such as "( applicable_errata = "RHBA-2017:2924" )" and this should send that too.

@jlsherrill thanks for finding this, fixed.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/details/erratum-content-hosts.controller.js [hunk 1]
```diff
@@ -69,7 +68,12 @@ angular.module('Bastion.errata').controller('ErratumContentHostsController',
             searchStatements = _.map(errataIds, function(errataId) {
                 return searchTerm + ' = "' + errataId + '"';
             });
-            return searchStatements.join(" or ");
+
+            hostSearchString = searchStatements.join(" or ");
+
+            IncrementalUpdate.setHostSearchString(hostSearchString);
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/details/erratum-content-hosts.controller.js]: this looks like its ignoring the user search on that page.  (its also ignoring the environment filter, maybe we should get rid of it?)

Comment (waldenraines) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/details/erratum-content-hosts.controller.js]: I believe that was also the case prior to my changes.

**Comment by waldenraines** [comment]
@jlsherrill this should be working as expected now (barring http://projects.theforeman.org/issues/21897).

**Comment by waldenraines** [comment]
> @jlsherrill this should be working as expected now (barring http://projects.theforeman.org/issues/21897).

Actually this was PEBCAK, this is ready to be reviewed.

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
@jlsherrill this is ready to test again, no idea why it is failing with some unrelated CV test.

**Comment by waldenraines** [comment]
[test]

**Comment by jlsherrill** [comment]
[test rails-5]

**Comment by jlsherrill** [comment]
It looks like the call to POST /api/v2/hosts/post_index   is no longer including the 'applicable_errata = FOO' search string, resulting in all hosts being shown for selection.

**Comment by waldenraines** [comment]
> It looks like the call to POST /api/v2/hosts/post_index is no longer including the 'applicable_errata = FOO' search string, resulting in all hosts being shown for selection.

So you do need that?

```
2018-01-10 10:08:58     jsherrill       also, i think i may have spread some fake news when i talked to you last
2018-01-10 10:09:03     jsherrill       you only need to pass whats in the search block
2018-01-10 10:09:15     jsherrill       not 'applicable_errata = \"RHEA-2012:0055\"'
2018-01-10 10:09:33     jsherrill       SAD!
```

**Comment by jlsherrill** [comment]
For the call to get the hosts yes (otherwise it will just return all
hosts), for the call to get available incremental updates, no (since the
errata are already being passed via other params).  Apologies if that was
not clear.

**Comment by waldenraines** [comment]
> For the call to get the hosts yes (otherwise it will just return all
> hosts), for the call to get available incremental updates, no (since the
> errata are already being passed via other params).  Apologies if that was
> not clear.

Are you sure that you have the latest?  I am seeing this:

```
{
	"organization_id": "1",
	"paged": true,
	"page": 1,
	"per_page": "20",
	"search": "( applicable_errata = \"RHSA-2017:3315\" or applicable_errata = \"RHBA-2017:3296\" or applicable_errata = \"RHBA-2017:3320\" )"
}
```

**Comment by jlsherrill** [comment]
Performing a search and then selecting all the hosts, and clicking 'apply to hosts' results in the search NOT being sent to the server, so it would apply to all hosts and not just what was searched for.

**Comment by waldenraines** [comment]
> Performing a search and then selecting all the hosts, and clicking 'apply to hosts' results in the search NOT being sent to the server, so it would apply to all hosts and not just what was searched for.

Thanks for noticing that, fixed.

**Comment by waldenraines** [comment]
[test hound]

**Comment by waldenraines** [comment]
[test]

**Comment by jlsherrill** [comment]
If i Don't select 'select all', i'm still seeing a search being sent up to the server as part of the available_incremental_updates call:

`search: "( applicable_errata = "RHEA-2012:0055" )`

This will cause it to perform like a 'select all'

**Comment by waldenraines** [comment]
@jlsherrill sorry for the back and forth on this but I think it works now.

**Pull request merged**